plugin/trino-hive/src/main/java/io/trino/plugin/hive/procedure/SyncPartitionMetadataProcedure.java

```diff
@@ -139,21 +141,45 @@ private void doSyncPartitionMetadata(ConnectorSession session, ConnectorAccessCo
 
             Location tableLocation = Location.of(table.getStorage().getLocation());
 
-            Set<String> partitionsInMetastore = metastore.getPartitionNames(schemaName, tableName)
+            Set<String> canonicalPartitionNamesInMetastore = metastore.getPartitionNames(schemaName, tableName)
                     .map(ImmutableSet::copyOf)
                     .orElseThrow(() -> new TableNotFoundException(schemaTableName));
+            String tableLocationDirectory = table.getStorage().getLocation().endsWith("/") ? table.getStorage().getLocation() : table.getStorage().getLocation() + "/";
```

Comment:
Can you use `tableLocation` here?
Also you could move this logic to `getPartitionNameFromPartitionLocation` as it's only used there.

Reply:
Slightly modified the code.

@pajaks  besides code cosmetics, do you have any concerns with regards to the bugfix?